File: app/src/main/java/com/bmn/bookfinder/activities/SplashScreenActivity.java
package com.bmn.bookfinder.activities;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.bmn.bookfinder.R;
import com.bmn.bookfinder.helpers.SharedPrefUtils;

public class SplashScreenActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        new Handler().postDelayed(this::updateUI, 2000);
    }

    private void updateUI() {
        if (SharedPrefUtils.loadFirstTimeUsedTimestamp(getApplicationContext()) != 0L) {
            startActivity(new Intent(getApplicationContext(), MainActivity.class));
        } else {
            startActivity(new Intent(getApplicationContext(), FirstUseSliderActivity.class));
        }
    }
}

File: app/src/main/java/com/bmn/bookfinder/activities/SplashScreenActivity.kt
package com.bmn.bookfinder.activities

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.bmn.bookfinder.R
import com.bmn.bookfinder.helpers.SharedPrefUtils

class SplashScreenActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        Handler().postDelayed({ updateUI() }, 2000)
    }

    private fun updateUI() {
        if (SharedPrefUtils.loadFirstTimeUsedTimestamp(applicationContext) != 0L) {
            startActivity(Intent(applicationContext, MainActivity::class.java))
        } else {
            startActivity(Intent(applicationContext, FirstUseSliderActivity::class.java))
        }
    }
}